### PR TITLE
crypto: added signify fuzzer

### PR DESCRIPTION
**Pull request by MariusVanDerWijden**

### CODE REVIEW

File: crypto/signify_fuzz.go [hunk 1]
```diff
@@ -0,0 +1,134 @@
...
+	err = SignifySignFile(tmpFile.Name(), tmpFile.Name()+".sig", testSecKey, "clé", "croissants")
+	if err != nil {
+		panic(err)
+	}
+	defer os.Remove(tmpFile.Name() + ".sig")
+
+	// if signify-openbsd is present, check the signature.
+	// signify-openbsd will be present in CI.
+	if runtime.GOOS == "linux" {
+		cmd := exec.Command("which", "signify")
```
Comment (gballet) [inline crypto/signify_fuzz.go:60]: would it somehow be possible to pass the executable name as some parameter to `Fuzz` ?

File: crypto/signify_fuzz.go [hunk 2]
```diff
@@ -0,0 +1,134 @@
...
+	// Create key and put it in correct format
+	tmpKey, err := ioutil.TempFile("", "")
+	cmd := exec.Command("signify", "-G", "-n", "-p", tmpKey.Name()+".pub", "-s", tmpKey.Name()+".sec")
+	if output, err := cmd.CombinedOutput(); err != nil {
+		panic(fmt.Sprintf("could not verify the file: %v, output: \n%s", err, output))
+	}
+	secKey, err := getKey(tmpKey.Name() + ".sec")
+	if err != nil {
+		panic(err)
+	}
```
Comment (gballet) [inline crypto/signify_fuzz.go:128]:
```suggestion
	}
	defer os.Remove(tmpKey.Name() + ".sec")
```

File: crypto/signify_fuzz.go [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+	pubKey, err := getKey(tmpKey.Name() + ".pub")
+	if err != nil {
+		panic(err)
+	}
```
Comment (gballet) [inline crypto/signify_fuzz.go:132]:
```suggestion
	}
	defer os.Remove(tmpKey.Name() + ".pub")
```

File: crypto/signify_fuzz.go [hunk 4]
```diff
@@ -0,0 +1,134 @@
...
+	defer tmpFile.Close()
+
+	testSecKey, testPubKey := createKeyPair()
+	// Create message
+	tmpFile.Write(data)
+	if err = tmpFile.Close(); err != nil {
+		panic(err)
+	}
+
+	err = SignifySignFile(tmpFile.Name(), tmpFile.Name()+".sig", testSecKey, "clé", "croissants")
```
Comment (gballet) [inline crypto/signify_fuzz.go:51]: would it be possible to generate the comments as random strings instead?

File: crypto/signify_fuzz.go (hunk 1 above)
Comment (MariusVanDerWijden) [reply crypto/signify_fuzz.go:60]: Not to fuzz, but we can pass it as an environment parameter

File: crypto/signify_fuzz.go (hunk 4 above)
Comment (MariusVanDerWijden) [reply crypto/signify_fuzz.go:51]: Sure

**Pull request merged**